packages/state-transition/src/cache/epochCache.ts

```diff
@@ -918,12 +918,22 @@ export class EpochCache {
    * balances are sampled to adjust the probability of the next selection (32 per epoch on average). So to invalidate
    * the prediction the effective of one of those 32 samples should change and change the random_byte inequality.
    */
-  getBeaconProposersNextEpoch(): ValidatorIndex[] {
+  async getBeaconProposersNextEpoch(): Promise<ValidatorIndex[]> {
     if (!this.proposersNextEpoch.computed) {
+      if (!this.nextShuffling) {
+        this.nextShuffling = (await this.shufflingCache?.get(this.nextEpoch, this.nextDecisionRoot)) ?? null;
+      }
+      if (!this.nextShuffling) {
+        throw new EpochCacheError({
+          code: EpochCacheErrorCode.NEXT_SHUFFLING_NOT_AVAILABLE,
+          epoch: this.nextEpoch,
+          decisionRoot: this.getShufflingDecisionRoot(this.nextEpoch),
+        });
+      }
       const indexes = computeProposers(
         this.config.getForkSeqAtEpoch(this.epoch + 1),
         this.proposersNextEpoch.seed,
-        this.getShufflingAtEpoch(this.nextEpoch),
```

Comment:
so I suppose it failed silently due to this function. Could you inspect all the consumers of this function 
and make sure it has no issue @matthewkeil ? for beacon-node we already have `shufflingCache` so no need to query here, could be in a separate PR

Reply:
> so I suppose it failed silently due to this function
yes, correct.  should be fine now after the refactor because the shuffling will be ready by time `getBeaconProposersNextEpoch` is called.

> Could you inspect all the consumers of this function and make sure it has no issue @matthewkeil ?
sure.  done and all are accounted for.  `getBeaconProposersNextEpoch` was not used anywhere before the PR nico wrote so it slipped through the cracks but all other consumers should be guarded against this issue

> for beacon-node we already have `shufflingCache` so no need to query here, could be in a separate PR
good idea!  refactored